src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -108,7 +153,50 @@ public ODataSerializerContext(ResourceContext resource, SelectExpandClause selec
         /// <summary>
         /// Gets or sets the <see cref="SelectExpandClause"/>.
         /// </summary>
-        public SelectExpandClause SelectExpandClause { get; set; }
+        public SelectExpandClause SelectExpandClause
+        {
+            get
+            {
+                // private backing field to be removed once public setter from ODataFeature is removed. 
+                if (_selectExpandClause != null && _isSelectExpandClauseSet)
```

Comment:
So, if I explicitly set the selectExpandClause to null, it will "unset it" and use the value from QueryOptions. Is this the desired behavior? If so, I'm not sure why we added isSelectExpandClauseSet, as opposed to just selectExpandClause != null.